### PR TITLE
chore: use `NoopBodiesDownloader` & `NoopHeaderDownloader` on `stage unwind` command

### PR DESCRIPTION
**Pull request by joshieDo**
Adds noop downloaders and uses them on `reth stage unwind`.

Stepping stone for a standalone unwind pipeline for [#8143](https://github.com/paradigmxyz/reth/pull/8143)

tested and unwinded with no issues

### CODE REVIEW

File: crates/net/downloaders/src/bodies/noop.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+use reth_interfaces::p2p::{
+    bodies::{downloader::BodyDownloader, response::BlockResponse},
+    error::{DownloadError, DownloadResult},
+};
+use reth_primitives::BlockNumber;
+use std::{ops::RangeInclusive, task::Poll};
+
+/// A [BodyDownloader] implementation that does nothing.
+#[derive(Debug, Default)]
+pub struct NoopBodiesDownloader;
```
Comment (mattsse) [inline crates/net/downloaders/src/bodies/noop.rs:11]:
```suggestion
#[non_exhaustive]
pub struct NoopBodiesDownloader;
```

File: crates/net/downloaders/src/bodies/noop.rs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+impl Stream for NoopBodiesDownloader {
+    type Item = Result<Vec<BlockResponse>, DownloadError>;
+
+    fn poll_next(
+        self: std::pin::Pin<&mut Self>,
+        _: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Option<Self::Item>> {
+        Poll::Ready(None)
```
Comment (mattsse) [inline crates/net/downloaders/src/bodies/noop.rs:26]: this should return pending

File: crates/net/downloaders/src/headers/noop.rs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+use reth_interfaces::p2p::headers::{
+    downloader::{HeaderDownloader, SyncTarget},
+    error::HeadersDownloaderError,
+};
+use reth_primitives::SealedHeader;
+use std::task::Poll;
+
+/// A [HeaderDownloader] implementation that does nothing.
+#[derive(Debug, Default)]
+pub struct NoopHeaderDownloader;
```
Comment (mattsse) [inline crates/net/downloaders/src/headers/noop.rs:11]:
```suggestion
#[non_exhaustive]
pub struct NoopHeaderDownloader;
```

File: crates/net/downloaders/src/headers/noop.rs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+impl Stream for NoopHeaderDownloader {
+    type Item = Result<Vec<SealedHeader>, HeadersDownloaderError>;
+
+    fn poll_next(
+        self: std::pin::Pin<&mut Self>,
+        _: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Option<Self::Item>> {
+        Poll::Ready(None)
```
Comment (mattsse) [inline crates/net/downloaders/src/headers/noop.rs:28]: should also return pending

**Review by mattsse — CHANGES_REQUESTED** [review]
supportive, smol nits

**Review by mattsse — APPROVED** [review]

File: crates/net/downloaders/src/headers/noop.rs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+impl Stream for NoopHeaderDownloader {
+    type Item = Result<Vec<SealedHeader>, HeadersDownloaderError>;
+
+    fn poll_next(
+        self: std::pin::Pin<&mut Self>,
+        _: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Option<Self::Item>> {
+        Poll::Pending
```
Comment (DaniPopes) [inline crates/net/downloaders/src/headers/noop.rs:29]: `Ready(None)`? Or panic, if this shouldn't be reached

File: crates/net/downloaders/src/bodies/noop.rs [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+impl Stream for NoopBodiesDownloader {
+    type Item = Result<Vec<BlockResponse>, DownloadError>;
+
+    fn poll_next(
+        self: std::pin::Pin<&mut Self>,
+        _: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Option<Self::Item>> {
+        Poll::Pending
```
Comment (DaniPopes) [inline crates/net/downloaders/src/bodies/noop.rs:27]: same

**Pull request merged**